--- lolcat-1.4.orig/setup.py	2024-03-03 22:53:24.000000000 +0100
+++ lolcat-1.4/setup.py	2024-08-03 16:28:34.207863200 +0200
@@ -46,11 +46,16 @@ Send a PR.
 
 setup(name='lolcat',
       version='1.4',
+      package='lolcat',
       description='Rainbows and unicorns!',
       long_description=LONG_DESCRIPTION,
       long_description_content_type='text/markdown',
       author='Wijnand Modderman-Lenstra',
       author_email='maze@pyth0n.org',
       url='https://github.com/tehmaze/lolcat/',
-      scripts=['lolcat'],
+      entry_points={
+          'console_scripts': [
+              'lolcat = lolcat.__main__:run',
+          ],
+      },
      )
