
Index: tclreadline-2.1.0/configure
===================================================================
--- tclreadline-2.1.0.orig/configure
+++ tclreadline-2.1.0/configure
@@ -2200,82 +2200,6 @@
 else
     echo "$ac_t""$TCL_INCLUDE_DIR/tcl.h" 1>&6
 fi
-
-
-# ---
-# check for the terminal library.
-# ---
-
-# Check whether --with-tlib-library or --without-tlib-library was given.
-if test "${with_tlib_library+set}" = set; then
-  withval="$with_tlib_library"
-  LIBS="$LIBS $withval"
-else
-  for tlib in ncurses termlib termcap curses; do
-	brk=yes
-	echo $ac_n "checking for tputs in -l$tlib""... $ac_c" 1>&6
-echo "configure:2139: checking for tputs in -l$tlib" >&5
-ac_lib_var=`echo $tlib'_'tputs | sed 'y%./+-%__p_%'`
-if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
-  echo $ac_n "(cached) $ac_c" 1>&6
-else
-  ac_save_LIBS="$LIBS"
-LIBS="-l$tlib  $LIBS"
-cat > conftest.$ac_ext <<EOF
-#line 2147 "configure"
-#include "confdefs.h"
-/* Override any gcc2 internal prototype to avoid an error.  */
-/* We use char because int might match the return type of a gcc2
-    builtin and then its argument prototype would still apply.  */
-char tputs();
-
-int main() {
-tputs()
-; return 0; }
-EOF
-if { (eval echo configure:2158: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=no"
-fi
-rm -f conftest*
-LIBS="$ac_save_LIBS"
-
-fi
-if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
-  echo "$ac_t""yes" 1>&6
-    ac_tr_lib=HAVE_LIB`echo $tlib | sed -e 's/^a-zA-Z0-9_/_/g' \
-    -e 'y/abcdefghijklmnopqrstuvwxyz/ABCDEFGHIJKLMNOPQRSTUVWXYZ/'`
-  cat >> confdefs.h <<EOF
-#define $ac_tr_lib 1
-EOF
-
-  LIBS="-l$tlib $LIBS"
-
-else
-  echo "$ac_t""no" 1>&6
-brk=no
-fi
-
-	if test $brk = "yes"; then
-	    HAVE_TLIB=yes
-	    break
-	fi
-    done
-    if test "x" = "x$HAVE_TLIB"; then
-	{ echo "configure: error: 
-	    no terminal library found. Please retry with the --with-tlib argument.
-	" 1>&2; exit 1; }
-    fi
-
-fi
-
-
-
 #--------------------------------------------------------------------
 #   Check for readline.h
 #--------------------------------------------------------------------
Index: tclreadline-2.1.0/configure.in
===================================================================
--- tclreadline-2.1.0.orig/configure.in
+++ tclreadline-2.1.0/configure.in
@@ -122,32 +122,6 @@
 else
     AC_MSG_RESULT($TCL_INCLUDE_DIR/tcl.h)
 fi
-
-
-# ---
-# check for the terminal library.
-# ---
-
-AC_ARG_WITH(tlib-library,
-    [  --with-tlib-library=DIR
-		  lib spec to tlib (e.g. '-L/usr/local/lib -lncurses')],
-    LIBS="$LIBS $withval",
-    for tlib in ncurses termlib termcap curses; do
-	brk=yes
-	AC_CHECK_LIB($tlib, tputs, , brk=no)
-	if test $brk = "yes"; then
-	    HAVE_TLIB=yes
-	    break
-	fi
-    done
-    if test "x" = "x$HAVE_TLIB"; then
-	AC_MSG_ERROR([
-	    no terminal library found. Please retry with the --with-tlib argument.
-	])
-    fi
-)
-
-
 #--------------------------------------------------------------------
 #   Check for readline.h
 #--------------------------------------------------------------------
