--- musicxml.orig/build/CMakeLists.txt	2021-10-11 19:45:29.174533600 +0200
+++ musicxml/build/CMakeLists.txt	2021-10-12 18:59:31.547101100 +0200
@@ -209,6 +209,10 @@ set (DOCDIR  ${CMAKE_INSTALL_PREFIX}/sha
 if (APPLE)
   set (BINDEST bin)
   set (LIBDEST lib)
+elseif (MINGW)
+  set (BINDEST bin)
+  set (LIBDEST lib)
+  set (DOCDIR  ${CMAKE_INSTALL_PREFIX}/share/doc/musicxml)
 elseif (WIN32)
   set (BINDEST bin)
   set (LIBDEST bin)
