--- origsrc/openldap-2.6.3/build/top.mk
+++ src/openldap-2.6.3/build/top.mk
@@ -91,6 +91,7 @@ NT_LTFLAGS_LIB = -no-undefined -avoid-ve
 NT_LTFLAGS_MOD = -no-undefined -avoid-version -rpath $(moduledir)
 UNIX_LTFLAGS_LIB = $(LTVERSION) -rpath $(libdir)
 UNIX_LTFLAGS_MOD = $(LTVERSION) -rpath $(moduledir)
+CYGWIN_LTFLAGS_LIB = -no-undefined $(LTVERSION) -rpath $(libdir)
 
 # libtool flags
 LTFLAGS     = $(@PLAT@_LTFLAGS)

--- origsrc/openldap-2.6.3/configure.ac
+++ src/openldap-2.6.3/configure.ac
@@ -3089,7 +3089,9 @@ dnl
 if test "$ac_cv_mingw32" = yes -o $ol_cv_msvc = yes ; then
 	PLAT=NT
 	SLAPD_MODULES_LDFLAGS=
+elif test "$ac_cv_cygwin" = yes ; then
+	PLAT=CYGWIN
 else
 	PLAT=UNIX
 fi

--- origsrc/openldap-2.6.3/libraries/liblber/Makefile.in
+++ src/openldap-2.6.3/libraries/liblber/Makefile.in
@@ -21,6 +21,9 @@ NT_OBJS = nt_err.lo
 UNIX_SRCS = stdio.c
 UNIX_OBJS = stdio.lo
 
+CYGWIN_SRCS = $(UNIX_SRCS)
+CYGWIN_OBJS = $(UNIX_OBJS)
+
 LIB_DEFS = -DLBER_LIBRARY
 
 SRCS= assert.c decode.c encode.c io.c bprint.c debug.c \
@@ -38,7 +41,8 @@ XLIBS = $(LIBRARY) $(LDAP_LIBLUTIL_A)
 XXLIBS = 
 NT_LINK_LIBS = $(AC_LIBS)
 UNIX_LINK_LIBS = $(AC_LIBS)
+CYGWIN_LINK_LIBS = $(UNIX_LINK_LIBS)
 ifneq (,$(OL_VERSIONED_SYMBOLS))
 	SYMBOL_VERSION_FLAGS=$(OL_VERSIONED_SYMBOLS)$(srcdir)/lber.map
 endif

--- origsrc/openldap-2.6.3/libraries/libldap/Makefile.in
+++ src/openldap-2.6.3/libraries/libldap/Makefile.in
@@ -58,7 +58,8 @@ XLIBS = $(LIBRARY) $(LDAP_LIBLBER_LA) $(
 XXLIBS = $(SECURITY_LIBS) $(LUTIL_LIBS)
 NT_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
 UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS) $(LTHREAD_LIBS)
+CYGWIN_LINK_LIBS = $(UNIX_LINK_LIBS)
 ifneq (,$(OL_VERSIONED_SYMBOLS))
 	SYMBOL_VERSION_FLAGS=$(OL_VERSIONED_SYMBOLS)$(srcdir)/ldap.map
 endif

--- origsrc/openldap-2.6.3/libraries/liblutil/Makefile.in
+++ src/openldap-2.6.3/libraries/liblutil/Makefile.in
@@ -24,6 +24,9 @@ NT_OBJS = ntservice.o slapdmsg.res
 UNIX_SRCS = detach.c
 UNIX_OBJS = detach.o
 
+CYGWIN_SRCS = $(UNIX_SRCS)
+CYGWIN_OBJS = $(UNIX_OBJS)
+
 XLIBS = $(LIBRARY) $(LDAP_LIBLBER_LA)
 
 SRCS	= base64.c entropy.c sasl.c signal.c hash.c passfile.c \

