--- libjpeg-turbo-2.0.5.orig/CMakeLists.txt	2020-06-19 06:40:20.000000000 +0200
+++ libjpeg-turbo-2.0.5/CMakeLists.txt	2020-08-04 19:22:37.459299000 +0200
@@ -631,14 +631,16 @@ if(WITH_TURBOJPEG)
       set_target_properties(turbojpeg-static PROPERTIES OUTPUT_NAME turbojpeg)
     endif()
 
-    add_executable(tjunittest-static tjunittest.c tjutil.c md5/md5.c
-      md5/md5hl.c)
-    target_link_libraries(tjunittest-static turbojpeg-static)
+    if(NOT MSYS)
+        add_executable(tjunittest-static tjunittest.c tjutil.c md5/md5.c
+          md5/md5hl.c)
+        target_link_libraries(tjunittest-static turbojpeg-static jpeg-static)
 
-    add_executable(tjbench-static tjbench.c tjutil.c)
-    target_link_libraries(tjbench-static turbojpeg-static)
-    if(UNIX)
-      target_link_libraries(tjbench-static m)
+        add_executable(tjbench-static tjbench.c tjutil.c)
+        target_link_libraries(tjbench-static turbojpeg-static jpeg-static)
+        if(UNIX)
+          target_link_libraries(tjbench-static m)
+        endif()
     endif()
   endif()
 endif()
@@ -654,7 +656,7 @@ else()
   set(DJPEG_BMP_SOURCES wrbmp.c wrtarga.c)
 endif()
 
-if(ENABLE_STATIC)
+if(ENABLE_STATIC AND NOT MSYS)
   add_executable(cjpeg-static cjpeg.c cdjpeg.c rdgif.c rdppm.c rdswitch.c
     ${CJPEG_BMP_SOURCES})
   set_property(TARGET cjpeg-static PROPERTY COMPILE_FLAGS ${COMPILE_FLAGS})
