diff -pur sbcl-2.2.10.orig/contrib/sb-mpfr/mpfr.lisp sbcl-2.2.10/contrib/sb-mpfr/mpfr.lisp
--- sbcl-2.2.10.orig/contrib/sb-mpfr/mpfr.lisp	2022-10-30 20:42:45.000000000 +0100
+++ sbcl-2.2.10/contrib/sb-mpfr/mpfr.lisp	2022-11-22 14:24:49.227370300 +0100
@@ -200,7 +200,7 @@
   (or (some #'try-load-shared-object
             #-(or win32 darwin) '("libmpfr.so" "libmpfr.so.4" "libmpfr.so.6")
             #+darwin '("libmpfr.dylib" "libmpfr.4.dylib" "libmpfr.6.dylib")
-            #+win32 '("mpfr.dll"))
+            #+win32 '("mpfr.dll" "libmpfr-6.dll"))
       (warn "MPFR was not loaded. This is likely because the shared library ~
              was not found.")))
 
diff -pur sbcl-2.2.10.orig/doc/manual/Makefile sbcl-2.2.10/doc/manual/Makefile
--- sbcl-2.2.10.orig/doc/manual/Makefile	2022-10-30 20:42:45.000000000 +0100
+++ sbcl-2.2.10/doc/manual/Makefile	2022-11-22 14:26:21.507365000 +0100
@@ -35,7 +35,7 @@ ifeq ($(DVIPS),)
 endif
 
 .PHONY: all
-all: $(ASDFTEXI) ps pdf info html
+all: $(ASDFTEXI) info html #ps pdf
 
 .PHONY: dist
 dist: html pdf
diff -pur sbcl-2.2.10.orig/doc/manual/missingdoc sbcl-2.2.10/doc/manual/missingdoc
--- sbcl-2.2.10.orig/doc/manual/missingdoc	1970-01-01 01:00:00.000000000 +0100
+++ sbcl-2.2.10/doc/manual/missingdoc	2022-11-22 10:10:00.314879800 +0100
@@ -0,0 +1,11 @@
+fun-sb-ext-posix-getenv.texinfo
+fun-sb-ext-process-kill.texinfo
+class-sb-bsd-sockets-local-socket.texinfo
+class-sb-bsd-sockets-local-abstract-socket.texinfo
+fun-sb-sprof-start-profiling.texinfo
+class-sb-posix-flock.texinfo
+class-sb-posix-passwd.texinfo
+class-sb-posix-termios.texinfo
+class-sb-posix-timeval.texinfo
+fun-sb-posix-readlink.texinfo
+fun-sb-posix-syslog.texinfo
