--- ocaml-4.10.0.orig/configure.ac	2020-02-21 15:22:37.000000000 +0100
+++ ocaml-4.10.0/configure.ac	2020-07-26 18:30:05.916707600 +0200
@@ -650,7 +650,7 @@ AS_CASE([$CC,$host],
         [x86_64-*-*], [flexdll_chain="mingw64"])
       flexlink="flexlink -chain $flexdll_chain -merge-manifest -stack 16777216"
       flexdir=`$flexlink -where | tr -d '\015'`
-      AS_IF([test -z "$flexdir"], [flexdir='$(ROOTDIR)/flexdll'])
+      AS_IF([test -z "$flexdir"], [flexdir='$(ROOTDIR)/flexdll'], [flexdir=$(realpath $flexdir/../lib/flexdll)])
       iflexdir="-I\"$flexdir\""
       mkexedebugflag="-link -g"])
     ostype="Win32"
