diff -pur ocaml-4.10.0.orig/runtime/win32.c ocaml-4.10.0/runtime/win32.c
--- ocaml-4.10.0.orig/runtime/win32.c	2020-02-21 15:22:37.000000000 +0100
+++ ocaml-4.10.0/runtime/win32.c	2020-07-26 17:27:31.997946700 +0200
@@ -228,8 +228,14 @@ void * caml_dlopen(wchar_t * libname, in
 {
   void *handle;
   int flags = (global ? FLEXDLL_RTLD_GLOBAL : 0);
+  size_t len = wcstombs(NULL, libname, 0);
+  char buf[len + 1];
+  wcstombs(buf, libname, len + 1);
+  for (int i = 0; buf[i] != '\0'; ++i)
+    if (buf[i] == '/')
+      buf[i] = '\\';
   if (!for_execution) flags |= FLEXDLL_RTLD_NOEXEC;
-  handle = flexdll_wdlopen(libname, flags);
+  handle = flexdll_dlopen(buf, flags);
   if ((handle != NULL) && ((caml_verb_gc & 0x100) != 0)) {
     flexdll_dump_exports(handle);
     fflush(stdout);
