diff -pur sox-14.4.2.orig/src/ladspa.c sox-14.4.2/src/ladspa.c
--- sox-14.4.2.orig/src/ladspa.c	2014-10-27 03:55:50.000000000 +0100
+++ sox-14.4.2/src/ladspa.c	2020-07-11 18:14:38.382772200 +0200
@@ -129,9 +129,16 @@ static int sox_ladspa_getopts(sox_effect
   path = getenv("LADSPA_PATH");
   if (path == NULL)
     path = LADSPA_PATH;
+  size_t len = strlen(l_st->name);
+  char dll_name[len + 4];
+  char so_name[len + 3];
 
+  strcpy(dll_name, l_st->name);
+  strcpy(so_name, l_st->name);
+  strcat(dll_name, ".dll");
+  strcat(so_name, ".so");
   if(lt_dlinit() || lt_dlsetsearchpath(path)
-      || (l_st->lth = lt_dlopenext(l_st->name)) == NULL) {
+     || (((l_st->lth = lt_dlopen(dll_name)) == NULL) && ((l_st->lth = lt_dlopen(so_name)) == NULL))) {
     lsx_fail("could not open LADSPA plugin %s", l_st->name);
     return SOX_EOF;
   }
