diff -pur sox-14.4.2.orig/src/Makefile.am sox-14.4.2/src/Makefile.am
--- sox-14.4.2.orig/src/Makefile.am	2014-10-30 04:29:54.000000000 +0100
+++ sox-14.4.2/src/Makefile.am	2020-07-12 14:23:19.192265900 +0200
@@ -12,7 +12,7 @@ if HAVE_LIBLTDL
 # is ignored when you specify a more specific one.
 # We want to version libsox and we are OK because they
 # have a more specific LDFLAGS that includes -version-info.
-AM_LDFLAGS = -avoid-version -module
+AM_LDFLAGS = -avoid-version -module @APP_LDFLAGS@
 AM_CPPFLAGS += -DPKGLIBDIR="\"$(pkglibdir)\""
 endif
 if HAVE_WIN32_LTDL
diff -pur sox-14.4.2.orig/src/optional-fmts.am sox-14.4.2/src/optional-fmts.am
--- sox-14.4.2.orig/src/optional-fmts.am	2014-10-06 04:02:30.000000000 +0200
+++ sox-14.4.2/src/optional-fmts.am	2020-07-12 14:39:53.879942200 +0200
@@ -25,7 +25,7 @@ else
 endif
 else
   libsox_fmt_amr_nb_la_SOURCES = amr-nb.c amr.h
-  libsox_fmt_amr_nb_la_LIBADD = libsox.la @AMRNB_LIBS@
+  libsox_fmt_amr_nb_la_LIBADD = @APP_LDFLAGS@ libsox.la @AMRNB_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_amr_nb.la
 endif
 endif
@@ -40,7 +40,7 @@ else
 endif
 else
   libsox_fmt_amr_wb_la_SOURCES = amr-wb.c amr.h
-  libsox_fmt_amr_wb_la_LIBADD = libsox.la @AMRWB_LIBS@
+  libsox_fmt_amr_wb_la_LIBADD = @APP_LDFLAGS@ libsox.la @AMRWB_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_amr_wb.la
 endif
 endif
@@ -55,7 +55,7 @@ else
 endif
 else
   libsox_fmt_ao_la_SOURCES = ao.c
-  libsox_fmt_ao_la_LIBADD = libsox.la @AO_LIBS@
+  libsox_fmt_ao_la_LIBADD = @APP_LDFLAGS@ libsox.la @AO_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_ao.la
 endif
 endif
@@ -85,7 +85,7 @@ else
 endif
 else
   libsox_fmt_flac_la_SOURCES = flac.c
-  libsox_fmt_flac_la_LIBADD = libsox.la @FLAC_LIBS@
+  libsox_fmt_flac_la_LIBADD = @APP_LDFLAGS@ libsox.la @FLAC_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_flac.la
 endif
 endif
@@ -100,7 +100,7 @@ else
 endif
 else
   libsox_fmt_gsm_la_SOURCES = gsm.c
-  libsox_fmt_gsm_la_LIBADD = libsox.la @GSM_LIBS@
+  libsox_fmt_gsm_la_LIBADD = @APP_LDFLAGS@ libsox.la @GSM_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_gsm.la
 endif
 endif
@@ -115,7 +115,7 @@ else
 endif
 else
   libsox_fmt_lpc10_la_SOURCES = lpc10.c
-  libsox_fmt_lpc10_la_LIBADD = libsox.la @LPC10_LIBS@
+  libsox_fmt_lpc10_la_LIBADD = @APP_LDFLAGS@ libsox.la @LPC10_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_lpc10.la
 endif
 endif
@@ -130,7 +130,7 @@ else
 endif
 else
   libsox_fmt_mp3_la_SOURCES = mp3.c mp3-util.h
-  libsox_fmt_mp3_la_LIBADD = libsox.la @MP3_LIBS@
+  libsox_fmt_mp3_la_LIBADD = @APP_LDFLAGS@ libsox.la @MP3_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_mp3.la
 endif
 endif
@@ -147,7 +147,7 @@ endif
 else
   libsox_fmt_opus_la_SOURCES = opus.c
   libsox_fmt_opus_la_CFLAGS = @OPUS_CFLAGS@
-  libsox_fmt_opus_la_LIBADD = libsox.la @OPUS_LIBS@
+  libsox_fmt_opus_la_LIBADD = @APP_LDFLAGS@ libsox.la @OPUS_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_opus.la
 endif
 endif
@@ -177,7 +177,7 @@ else
 endif
 else
   libsox_fmt_pulseaudio_la_SOURCES = pulseaudio.c
-  libsox_fmt_pulseaudio_la_LIBADD = libsox.la @PULSEAUDIO_LIBS@
+  libsox_fmt_pulseaudio_la_LIBADD = @APP_LDFLAGS@ libsox.la @PULSEAUDIO_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_pulseaudio.la
 endif
 endif
@@ -192,7 +192,7 @@ else
 endif
 else
   libsox_fmt_waveaudio_la_SOURCES = waveaudio.c
-  libsox_fmt_waveaudio_la_LIBADD = libsox.la @WAVEAUDIO_LIBS@
+  libsox_fmt_waveaudio_la_LIBADD = @APP_LDFLAGS@ libsox.la @WAVEAUDIO_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_waveaudio.la
 endif
 endif
@@ -238,7 +238,7 @@ else
 endif
 else
   libsox_fmt_vorbis_la_SOURCES = vorbis.c
-  libsox_fmt_vorbis_la_LIBADD = libsox.la @OGG_VORBIS_LIBS@
+  libsox_fmt_vorbis_la_LIBADD = @APP_LDFLAGS@ libsox.la @OGG_VORBIS_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_vorbis.la
 endif
 endif
@@ -253,7 +253,7 @@ else
 endif
 else
   libsox_fmt_wavpack_la_SOURCES = wavpack.c
-  libsox_fmt_wavpack_la_LIBADD = libsox.la @WAVPACK_LIBS@
+  libsox_fmt_wavpack_la_LIBADD = @APP_LDFLAGS@ libsox.la @WAVPACK_LIBS@
   pkglib_LTLIBRARIES += libsox_fmt_wavpack.la
 endif
 endif
@@ -271,7 +271,7 @@ endif
 else
 libsox_fmt_sndfile_la_SOURCES = sndfile.c
 libsox_fmt_sndfile_la_CFLAGS = @WARN_CFLAGS@ @SNDFILE_CFLAGS@
-libsox_fmt_sndfile_la_LIBADD = libsox.la @SNDFILE_LIBS@
+libsox_fmt_sndfile_la_LIBADD = @APP_LDFLAGS@ libsox.la @SNDFILE_LIBS@
 libsox_fmt_caf_la_SOURCES = caf.c sndfile.c
 libsox_fmt_caf_la_CFLAGS = @WARN_CFLAGS@ @SNDFILE_CFLAGS@
 libsox_fmt_caf_la_LIBADD = libsox.la @SNDFILE_LIBS@
