--- popt-1.16/poptconfig.c.orig 2009-05-20 17:18:07 +0400
+++ popt-1.16/poptconfig.c      2012-03-31 08:50:54 +0400
@@ -141,17 +141,23 @@ static int poptGlob(/*@unused@*/ UNUSED(
 int poptSaneFile(const char * fn)
 {
     struct stat sb;
+#ifndef __MINGW32__
     uid_t uid = getuid();
+#endif	
 
     if (stat(fn, &sb) == -1)
 	return 1;
+#ifndef __MINGW32__	
     if ((uid_t)sb.st_uid != uid)
 	return 0;
+#endif	
     if (!S_ISREG(sb.st_mode))
 	return 0;
 /*@-bitwisesigned@*/
+#ifndef __MINGW32__	
     if (sb.st_mode & (S_IWGRP|S_IWOTH))
 	return 0;
+#endif	
 /*@=bitwisesigned@*/
     return 1;
 }
